third_party/nvfuser/python_tests/test_python_frontend.py

```diff
@@ -726,6 +726,39 @@ def test_python_version_API(self):
         from packaging.version import Version
         self.assertTrue(version() > '0.0.0')
         self.assertTrue(version() > Version('0.0.0'))
+    
+    def test_def_and_sched_func_errors (self) :
+        inputs = [
+            torch.randn(4, 4, 4, device='cuda'),
+        ]
+
+        class DefError(FusionDefinition):
+            def definition(self) :
+                t0 = self.from_pytorch(inputs[0])
+                t1 = self.ops.tanh(t0)
+                self.add_output(t1)
+                self.sched.merge(t1, 1)
+
+        try:
+            fd = DefError()
+            out = fd.execute(inputs)
+        except RuntimeError:
+            pass
+        
+        class SchedError(FusionDefinition):
+            def definition(self) :
+                self.t0 = self.from_pytorch(inputs[0])
+                self.t1 = self.ops.tanh(self.t0)
+                self.add_output(self.t1)
+
+            def schedule(self) :
+                self.t2 = self.ops.relu(self.t1)
+
+        try:
+            fd = SchedError()
+            out = fd.execute(inputs)
+        except RuntimeError:
+            pass
```

Comment:
Can we add a functional test? looks like there's only testing for failures.

Reply:
I added a functional test.